Source/WebCore/css/CSSPrimitiveValueMappings.h

```diff
@@ -1389,6 +1389,48 @@ template<> constexpr WhiteSpace fromCSSValueID(CSSValueID valueID)
     return WhiteSpace::Normal;
 }
 
+constexpr CSSValueID toCSSValueID(WhiteSpaceCollapse e)
+{
+    switch (e) {
+    case WhiteSpaceCollapse::Collapse:
+        return CSSValueCollapse;
+    case WhiteSpaceCollapse::Discard:
+        return CSSValueDiscard;
+    case WhiteSpaceCollapse::Preserve:
+        return CSSValuePreserve;
+    case WhiteSpaceCollapse::PreserveBreaks:
+        return CSSValuePreserveBreaks;
+    case WhiteSpaceCollapse::PreserveSpaces:
+        return CSSValuePreserveSpaces;
+    case WhiteSpaceCollapse::BreakSpaces:
+        return CSSValueBreakSpaces;
+    }
+    ASSERT_NOT_REACHED_UNDER_CONSTEXPR_CONTEXT();
+    return CSSValueInvalid;
+}
+
+template<> constexpr WhiteSpaceCollapse fromCSSValueID(CSSValueID valueID)
+{
+    switch (valueID) {
+    case CSSValueCollapse:
+        return WhiteSpaceCollapse::Collapse;
+    case CSSValueDiscard:
+        return WhiteSpaceCollapse::Discard;
+    case CSSValuePreserve:
+        return WhiteSpaceCollapse::Preserve;
+    case CSSValuePreserveBreaks:
+        return WhiteSpaceCollapse::PreserveBreaks;
+    case CSSValuePreserveSpaces:
+        return WhiteSpaceCollapse::PreserveSpaces;
+    case CSSValueBreakSpaces:
+        return WhiteSpaceCollapse::BreakSpaces;
+    default:
+        break;
+    }
+    ASSERT_NOT_REACHED_UNDER_CONSTEXPR_CONTEXT();
+    return WhiteSpaceCollapse::Collapse;
+}
+
```

Comment:
Pretty sure you can use the macro here:

```suggestion
#define TYPE WhiteSpaceCollapse
#define FOR_EACH(CASE) CASE(Collapse) CASE(Discard) CASE(Preserve) CASE(PreserveBreaks) CASE(PreserveSpaces) CASE(BreakSpaces)
DEFINE_TO_FROM_CSS_VALUE_ID_FUNCTIONS
#undef TYPE
#undef FOR_EACH

```